### PR TITLE
Directory scan

### PR DESCRIPTION
**Pull request by willmcgugan**
### What this PR solves
- Implements new feature to expose remote directory
 
### How it is done
- Adds a thread to scan directories on regular intervals, and a new m2m service to do it on-demand

### Review
- [ ] Ready for review

### CODE REVIEW

File: dataplicity/scan_directory.py
Comment (willmcgugan) [inline dataplicity/scan_directory.py:34]: This code uses scandir which is available on Python3.5+ or as a backport.

Scandir makes it quiet efficient, and there is no code-level recursion. It will also handle recursive links which otherwise might turn this in to an infinite loop.

File: dataplicity/m2mmanager.py
Comment (willmcgugan) [inline dataplicity/m2mmanager.py:194]: Doesn't actually need / use m2m. The service just schedules a scan in another thread. Can't do it in the main thread as a scan could take multiple seconds and we don't want to hold up terminals etc.

**Review by toudi — APPROVED** [review]
I like how it's all separated - the thread from the actual scanning logic. good job!

**Comment by willmcgugan** [comment]
Thanks @toudi I'm going to merge this in to a feature branch, as it will be a while before I can merge this in to master.

**Pull request merged**
